HW3/Program.cs

```diff
@@ -0,0 +1,122 @@
+﻿using HW3.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace HW3
+{
+    internal class Program
+    {
+        private const string usersFile = "Users.txt";
+
+        private static bool ConsoleLogin()
+        {
+            try
+            {
+                string username = ConsoleReadLineWithTextPrompt("Username:\t");
+
+                string password = ConsoleReadLineWithTextPrompt("Password:\t");
+
+                return VerifyCredentials(username, password);
+            }
+            catch (UsersNotFoundException ex)
+            {
+                Console.WriteLine($"Couldn't load user credentials. Please contact @Kaisinel: {ex.Message}");
+                return false;
+            }
+        }
+
+        private static string ConsoleReadLineWithTextPrompt(string promptText)
+        {
+            Console.Write(promptText);
+            string username = Console.ReadLine();
+            return username;
+        }
+
+        private static void ConsoleRegisterAccount()
+        {
+            Dictionary<string, string> users;
+            try
+            {
+                users = IO.Read.UserCredentials(usersFile);
+            }
+            catch (UsersNotFoundException ex)
+            {
+                Console.WriteLine($"Couldn't load user credentials. Please contact @Kaisinel: {ex.Message}");
+                return;
+            }
+
+            do
+            {
+                string username = ConsoleReadLineWithTextPrompt("Username:\t");
```

Comment:
Nit: In some places you have newlines where I would not.  Try to avoid making your program look like you double spaced it.  Spaces in code are valuable markers if you use them judiciously.  If you use too many then you lose the effect.

Reply:
I know I'm not consistent with new lines... I try to seperate everything with 1 line (except for fields wich are nicely grouped together). I use CodeMaid to get rid of any extra lines but sometimes I clutter everything together and simply forget that CM only removes extra lines but does not insert them. Maybe I should reconsider this habit...